src/generators/android/templates/typewriterUtils.java.hbs

```diff
@@ -0,0 +1,45 @@
+package com.segment.generated;
+
+import com.segment.analytics.Options;
+import java.util.*;
+import java.lang.String;
+import androidx.annotation.Nullable;
+import androidx.annotation.NonNull;
+
+public final class TypewriterUtils {
+  protected static Map<String, String> typewriterCtx;
+
+  static {
+    typewriterCtx = new HashMap<>();
+    typewriterCtx.put("version", "7.0.1");
+    typewriterCtx.put("language", "java");
+  }
+
+  protected static Options addTypewriterContext(final @NonNull Options options){
+    options.putContext("typewriter", typewriterCtx);
+    return options;
+  }
+
+  protected static Options addTypewriterContext(){
+    Options typewriterContext = new Options();
+    typewriterContext.putContext("typewriter", typewriterCtx);
+    return typewriterContext;
+  }
+
+  protected static List<?> serialize(final @NonNull List<?> props){
+    List p = new ArrayList<>();
+    if(props instanceof List){
```

Comment:
Do we need this check? Isn't it always a `List`?

Reply:
Ah yeah my modifier is off here for the `props` arg - should be `@Nullable`. The initial check 
`props instanceof List` guards against that.

Reply:
ah, makes sense! could we do `if (props != null)` then to clarify it?